lib/Tools/circt-verilog-lsp-server/VerilogServerImpl/VerilogServer.cpp

```diff
@@ -976,53 +984,52 @@ static llvm::lsp::Range getRange(const mlir::FileLineColRange &fileLoc) {
       llvm::lsp::Position(fileLoc.getEndLine(), fileLoc.getEndColumn()));
 }
 
-// LSP (0-based line, UTF-16 character) -> byte offset into UTF-8 buffer.
-static inline uint32_t lspPositionToOffset(const slang::SourceManager &sm,
-                                           const slang::BufferID &buf,
-                                           const llvm::lsp::Position &pos) {
-  std::string_view text = sm.getSourceText(buf);
-
-  // 1) Find start-of-line (clamp to EOF if line is past end)
-  size_t lineStart = 0;
-  for (int i = 0; i < pos.line && lineStart < text.size(); ++i) {
-    size_t nl = text.find('\n', lineStart);
-    if (nl == std::string_view::npos) {
-      lineStart = text.size();
-      break;
+/// Build a vector of line start offsets (0-based).
+void VerilogDocument::computeLineOffsets(std::string_view text) {
```

Comment:
Is it different from SourceManager::computeLineOffsets? https://github.com/MikePopoloski/slang/blob/d9ac86a723653474548737f55384b0d99f447d92/source/text/SourceManager.cpp#L758C6-L758C39